Java-Basics.md

```diff
@@ -0,0 +1,442 @@
+# Java Basics
+
+## Core Concepts
+
+### The Java Virtual Machine (JVM) 
+
+Java belongs to a family of languages called **Compiled Languages** (https://en.wikipedia.org/wiki/Compiled_language). Any code written in such a language needs to be converted (compiled) to an intermediate form that can then be understood by the host platform (the OS/platform in which the code runs). For Java, this intermediate form is called **Bytecode** which is then interpreted by a runtime called a Java Virtual Machine (JVM). Think of JVM (https://docs.oracle.com/javase/specs/jvms/se7/html/) as a piece of software that does the hard work of running your Java code. It takes care of memory allocation, thread management, garbage collection and so much more. Apart from Java, it also supports (read: able to run) code written in languages such as Groovy, Scala etc.
```

Comment:
Begin a new paragraph from

> For Java, this intermediate form is called[...]

It gets much easier to read, once you break your text in smaller paragraphs. It's called [breaking the wall of text](http://tweakyourbiz.com/marketing/2012/08/10/avoiding-the-wall-of-text-when-creating-content/).